src/provider.rs

```diff
@@ -0,0 +1,87 @@
+use crate::dc_tools::EmailAddress;
+
+#[derive(Debug, Copy, Clone, PartialEq, ToPrimitive)]
+#[repr(u8)]
+pub enum Status {
+    OK = 1,
+    PREPARATION = 2,
+    BROKEN = 3,
+}
+
+#[derive(Debug)]
+#[repr(u8)]
+pub enum ServerType {
+    SMTP = 1,
+    IMAP = 2,
+}
+
+#[derive(Debug)]
+#[repr(u8)]
+pub enum ServerSocket {
+    STARTTLS = 1,
+    SSL = 2,
+}
+
+#[derive(Debug)]
+pub struct Server {
+    pub stype: ServerType,
+    pub socket: ServerSocket,
+    pub port: u16,
+    pub server: &'static str,
+    pub username: &'static str,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Provider {
+    pub domains: &'static str,
+    pub status: Status,
+    pub before_login_hint: &'static str,
+    pub overview_page: &'static str,
+}
+
+// TODO: the database will be auto-generated from the provider-db
+const DATABASE: [Provider; 3] = [
+    Provider {
+        domains: "nauta.cu",
+        status: Status::OK,
+        before_login_hint: "",
+        overview_page: "",
+    },
+    Provider {
+        domains: "outlook.com hotmail.com live.com",
```

Comment:
it would make for a nicer rust api, if this was a `Vec<>`

Reply:
i failed to get this working with static data.

Reply:
yeah, allocating a vec requires lazy_static